diff -up v8/src/allocation.cc.allocation v8/src/allocation.cc
--- v8/src/allocation.cc.allocation	2009-11-18 22:37:12.000000000 -0500
+++ v8/src/allocation.cc	2009-11-18 22:38:33.000000000 -0500
@@ -88,8 +88,8 @@ char* StrDup(const char* str) {
 }
 
 
-char* StrNDup(const char* str, int n) {
-  int length = StrLength(str);
+char* StrNDup(const char* str, size_t n) {
+  size_t length = StrLength(str);
   if (n < length) length = n;
   char* result = NewArray<char>(length + 1);
   memcpy(result, str, length * kCharSize);
diff -up v8/src/d8-readline.cc.allocation v8/src/d8-readline.cc
--- v8/src/d8-readline.cc.allocation	2009-11-18 15:09:01.000000000 -0500
+++ v8/src/d8-readline.cc	2009-11-18 22:37:58.000000000 -0500
@@ -32,6 +32,7 @@
 
 
 #include "d8.h"
+#include "allocation.h"
 
 
 // There are incompatibilities between different versions and different
@@ -103,7 +104,7 @@ char* ReadLineEditor::CompletionGenerato
   static unsigned current_index;
   static Persistent<Array> current_completions;
   if (state == 0) {
-    i::SmartPointer<char> full_text(i::StrNDup(rl_line_buffer, rl_point));
+    i::SmartPointer<char> full_text(i::StrNDup(rl_line_buffer, (size_t) rl_point));
     HandleScope scope;
     Handle<Array> completions =
       Shell::GetCompletions(String::New(text), String::New(*full_text));
