diff -ur v8-/src/d8-readline.cc v8/src/d8-readline.cc
--- v8-/src/d8-readline.cc	2010-03-09 20:28:47.000000000 -0700
+++ v8/src/d8-readline.cc	2010-03-09 20:52:00.000000000 -0700
@@ -26,7 +26,7 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
-#include <cstdio>  // NOLINT
+#include <stdio.h>  // NOLINT
 #include <readline/readline.h> // NOLINT
 #include <readline/history.h> // NOLINT
 
