--- v8-3.6.6.7/SConstruct~	2011-11-14 08:20:47.000000000 +0200
+++ v8-3.6.6.7/SConstruct	2011-11-14 08:24:01.929450494 +0200
@@ -1411,6 +1411,7 @@
   # Build the object files by invoking SCons recursively.
   d8_env = Environment(tools=tools)
   d8_env.Replace(**context.flags['d8'])
+  d8_env.Prepend(LIBS=['v8'], LIBPATH='.')
   (object_files, shell_files, mksnapshot, preparser_files) = env.SConscript(
     join('src', 'SConscript'),
     build_dir=join('obj', target_id),
