diff -ur v8-/src/allocation.h v8/src/allocation.h
--- v8-/src/allocation.h	2010-03-09 20:28:45.000000000 -0700
+++ v8/src/allocation.h	2010-03-09 21:07:04.000000000 -0700
@@ -124,7 +124,7 @@
 // and StrNDup uses new and calls the FatalProcessOutOfMemory handler
 // if allocation fails.
 char* StrDup(const char* str);
-char* StrNDup(const char* str, int n);
+char* StrNDup(const char* str, size_t n);
 
 
 // Allocation policy for allocating in the C free store using malloc
