--- v8-3.6.6.7/SConstruct	2011-11-14 08:19:38.420636802 +0200
+++ v8-3.6.6.7/SConstruct	2011-11-14 13:20:28.692835719 +0200
@@ -1308,17 +1308,17 @@
   suffix = SUFFIXES[target_id]
   library_name = 'v8' + suffix
   preparser_library_name = 'v8preparser' + suffix
-  version = GetVersion()
-  if context.options['soname'] == 'on':
-    # When building shared object with SONAME version the library name.
-    library_name += '-' + version
+  version = GetVersion().split('.')[0]
 
   # Generate library SONAME if required by the build.
   if context.options['soname'] == 'on':
     soname = GetSpecificSONAME()
     if soname == '':
-      soname = 'lib' + library_name + '.so'
-    env['SONAME'] = soname
+      soname = 'lib' + library_name + '.so.' + version
+      prepsoname = 'lib' + preparser_library_name + '.so.' + version
+  else:
+    soname = None
+    prepsoname = None
 
   # Build the object files by invoking SCons recursively.
   (object_files, shell_files, mksnapshot, preparser_files) = env.SConscript(
@@ -1434,9 +1434,9 @@
     # This should not affect any other platforms.
     pdb_name = library_name + '.dll.pdb'
     pdb_name = library_name + '.dll.pdb'
-    library = env.SharedLibrary(library_name, object_files, PDB=pdb_name)
+    library = env.SharedLibrary(library_name, object_files, PDB=pdb_name, SONAME=soname)
     preparser_pdb_name = preparser_library_name + '.dll.pdb';
-    preparser_soname = 'lib' + preparser_library_name + '.so';
+    preparser_soname = 'lib' + preparser_library_name + '.so.' + version;
     preparser_library = env.SharedLibrary(preparser_library_name,
                                           preparser_files,
                                           PDB=preparser_pdb_name,
