diff -ur v8-3.2.10.6-o/SConstruct v8-3.2.10.6/SConstruct
--- v8-3.2.10.6-o/SConstruct	2011-05-14 00:38:09.000000000 -0600
+++ v8-3.2.10.6/SConstruct	2011-05-14 00:38:49.000000000 -0600
@@ -1325,15 +1325,12 @@
   library_name = 'v8' + suffix
   preparser_library_name = 'v8preparser' + suffix
   version = GetVersion()
-  if context.options['soname'] == 'on':
-    # When building shared object with SONAME version the library name.
-    library_name += '-' + version
 
   # Generate library SONAME if required by the build.
   if context.options['soname'] == 'on':
     soname = GetSpecificSONAME()
     if soname == '':
-      soname = 'lib' + library_name + '.so'
+      soname = 'lib' + library_name + '.so.' + version
     env['SONAME'] = soname
 
   # Build the object files by invoking SCons recursively.
